pbm/node.go

```diff
@@ -106,6 +106,28 @@ func (n *Node) GetInfo() (*NodeInfo, error) {
 	return i, nil
 }
 
+// MaxDBSize returns the size of the largest database on replicaset.
+// It is the of files on disk in bytes
+func (n *Node) MaxDBSize() (int, error) {
+	i := &struct {
+		Databases []struct {
+			SizeOnDisk int `bson:"sizeOnDisk"`
+		} `bson:"databases"`
+	}{}
+	err := n.cn.Database(DB).RunCommand(n.ctx, bson.D{{"listDatabases", 1}}).Decode(i)
+	if err != nil {
+		return 0, errors.Wrap(err, "run mongo command listDatabases")
+	}
+
+	sized := 0
+	for _, db := range i.Databases {
+		if db.SizeOnDisk > sized {
```

Comment:
Proportional to the amount of WT blocks for deleted documents that might exist in the data files this may be larger than the dumped file. But other than that this is will be an accurate prediction of the dump size with s2 compression.

Reply:
To get the uncompressed size we'd have to run a dbStats command and look at the "dataSize" property. This has to be called database-by-database.